### ISSUE TITLE
outputs = self.fn() throws "MemoryError: alloc failed" on a PC which has more memory

### ISSUE THREAD

**Issue by munichong** [Theano/Theano#5026]
I am using Theano 0.8.2

My RNN is implemented by Keras with theano backend. It works fine on my Linux laptop with 16G RAM. Now, I move the program to my Windows desktop with 32G RAM, which is supposed to be more powerful. On both computers, I use Eclipse Pydev to run the program.

I run the same program that runs fine on the laptop. But I receive an error as below. It seems that it is a memory error. But I feel confused because the desktop should have more memory. I have tried to switch from CPU to GPU (6G) on the desktop. It does not help.  

Also, at the end of the error, it says  "TotalSize: 260991796.0 Byte(s) 0.243 GB, TotalSize inputs: 22558484.0 Byte(s) 0.021 GB". I guess it is the total size that my program requests. But it looks not very big. 26G out of the 32G RAM are idle. It should fit.

I am thinking if I need to manually allocate more memory or something to the program.

```
Using Theano backend.

Build model...

****** Iterating over each batch of the training data ******
Traceback (most recent call last):
  File "I:\Anaconda3\lib\site-packages\theano\compile\function_module.py", line 859, in __call__
    outputs = self.fn()
MemoryError: alloc failed

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "J:\git\DwellTimePrediction\DwellTimePrediction\Scenario3\models_training.py", line 107, in <module>
    loss_rnn = rnn.train_on_batch({'dep_input':X_batch_dep, 'ctx_input':X_batch_ctx}, y_batch)
  File "I:\Anaconda3\lib\site-packages\keras\engine\training.py", line 1219, in train_on_batch
    outputs = self.train_function(ins)
  File "I:\Anaconda3\lib\site-packages\keras\backend\theano_backend.py", line 672, in __call__
    return self.function(*inputs)
  File "I:\Anaconda3\lib\site-packages\theano\compile\function_module.py", line 871, in __call__
    storage_map=getattr(self.fn, 'storage_map', None))
  File "I:\Anaconda3\lib\site-packages\theano\gof\link.py", line 314, in raise_with_op
    reraise(exc_type, exc_value, exc_trace)
  File "I:\Anaconda3\lib\site-packages\six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "I:\Anaconda3\lib\site-packages\theano\compile\function_module.py", line 859, in __call__
    outputs = self.fn()
MemoryError: alloc failed
Apply node that caused the error: Alloc(TensorConstant{(1, 1, 1) of 0.0}, TensorConstant{100}, Elemwise{Composite{(((i0 // i1) * i2) // i3)}}.0, TensorConstant{200})
Toposort index: 167
Inputs types: [TensorType(float32, (True, True, True)), TensorType(int64, scalar), TensorType(int64, scalar), TensorType(int64, scalar)]
Inputs shapes: [(1, 1, 1), (), (), ()]
Inputs strides: [(4, 4, 4), (), (), ()]
Inputs values: [array([[[ 0.]]], dtype=float32), array(100, dtype=int64), array(-5334, dtype=int64), array(200, dtype=int64)]
Outputs clients: [[IncSubtensor{Inc;:int64:}(Alloc.0, Subtensor{::int64}.0, Constant{100}), IncSubtensor{InplaceInc;int64::}(Alloc.0, IncSubtensor{Inc;:int64:}.0, Constant{0}), forall_inplace,cpu,grad_of_scan_fn}(TensorConstant{100}, Elemwise{tanh}.0, Subtensor{int64:int64:int64}.0, Alloc.0, Elemwise{Composite{(i0 - sqr(i1))}}.0, Subtensor{int64:int64:int64}.0, Subtensor{int64:int64:int64}.0, Subtensor{int64:int64:int64}.0, Alloc.0, Alloc.0, TensorConstant{100}, TensorConstant{100}, TensorConstant{100}, TensorConstant{100}, TensorConstant{100}, lstm_2_U_o, lstm_2_U_f, lstm_2_U_i, lstm_2_U_c, InplaceDimShuffle{1,0}.0, InplaceDimShuffle{1,0}.0, InplaceDimShuffle{1,0}.0, InplaceDimShuffle{1,0}.0)]]

Backtrace when the node is created(use Theano flag traceback.limit=N to make it longer):
  File "I:\Anaconda3\lib\site-packages\theano\gradient.py", line 1279, in access_grad_cache
    term = access_term_cache(node)[idx]
  File "I:\Anaconda3\lib\site-packages\theano\gradient.py", line 973, in access_term_cache
    output_grads = [access_grad_cache(var) for var in node.outputs]
  File "I:\Anaconda3\lib\site-packages\theano\gradient.py", line 973, in <listcomp>
    output_grads = [access_grad_cache(var) for var in node.outputs]
  File "I:\Anaconda3\lib\site-packages\theano\gradient.py", line 1279, in access_grad_cache
    term = access_term_cache(node)[idx]
  File "I:\Anaconda3\lib\site-packages\theano\gradient.py", line 973, in access_term_cache
    output_grads = [access_grad_cache(var) for var in node.outputs]
  File "I:\Anaconda3\lib\site-packages\theano\gradient.py", line 973, in <listcomp>
    output_grads = [access_grad_cache(var) for var in node.outputs]
  File "I:\Anaconda3\lib\site-packages\theano\gradient.py", line 1279, in access_grad_cache
    term = access_term_cache(node)[idx]
  File "I:\Anaconda3\lib\site-packages\theano\gradient.py", line 1113, in access_term_cache
    input_grads = node.op.grad(inputs, new_output_grads)

Debugprint of the apply node: 
Alloc [id A] <TensorType(float32, 3D)> ''   
 |TensorConstant{(1, 1, 1) of 0.0} [id B] <TensorType(float32, (True, True, True))>
 |TensorConstant{100} [id C] <TensorType(int64, scalar)>
 |Elemwise{Composite{(((i0 // i1) * i2) // i3)}} [id D] <TensorType(int64, scalar)> ''   
 | |Elemwise{Mul}[(0, 1)] [id E] <TensorType(int64, scalar)> ''   
 | | |TensorConstant{30000} [id F] <TensorType(int64, scalar)>
 | | |Elemwise{Composite{((i0 * i1) // i2)}}[(0, 0)] [id G] <TensorType(int64, scalar)> ''   
 | |   |Elemwise{Composite{((i0 * i1 * i2) // i3)}}[(0, 1)] [id H] <TensorType(int64, scalar)> ''   
 | |   | |Elemwise{Composite{Switch(EQ(i0, i1), i2, i0)}} [id I] <TensorType(int64, scalar)> ''   
 | |   | | |Shape_i{0} [id J] <TensorType(int64, scalar)> ''   
 | |   | | | |dep_input [id K] <TensorType(float32, matrix)>
 | |   | | |TensorConstant{-1} [id L] <TensorType(int8, scalar)>
 | |   | | |Elemwise{Composite{((i0 * i1 * i2) // (-(i1 * i2 * i0)))}} [id M] <TensorType(int64, scalar)> ''   
 | |   | |   |Shape_i{1} [id N] <TensorType(int64, scalar)> ''   
 | |   | |   | |embedding_1_W [id O] <TensorType(float32, matrix)>
 | |   | |   |Shape_i{0} [id J] <TensorType(int64, scalar)> ''   
 | |   | |   |Shape_i{1} [id P] <TensorType(int64, scalar)> ''   
 | |   | |     |dep_input [id K] <TensorType(float32, matrix)>
 | |   | |Elemwise{Composite{Switch(EQ(i0, i1), i2, i0)}}[(0, 0)] [id Q] <TensorType(int64, scalar)> ''   
 | |   | | |Shape_i{1} [id P] <TensorType(int64, scalar)> ''   
 | |   | | |TensorConstant{-1} [id L] <TensorType(int8, scalar)>
 | |   | | |Elemwise{Composite{((i0 * i1 * i2) // (-(i1 * i2 * i0)))}} [id M] <TensorType(int64, scalar)> ''   
 | |   | |Elemwise{Add}[(0, 0)] [id R] <TensorType(int64, scalar)> ''   
 | |   | | |Elemwise{Composite{Switch(EQ(i0, i1), i2, i0)}}[(0, 0)] [id S] <TensorType(int64, scalar)> ''   
 | |   | | | |Shape_i{1} [id N] <TensorType(int64, scalar)> ''   
 | |   | | | |TensorConstant{-1} [id L] <TensorType(int8, scalar)>
 | |   | | | |Elemwise{Composite{((i0 * i1 * i2) // (-(i1 * i2 * i0)))}} [id M] <TensorType(int64, scalar)> ''   
 | |   | | |Shape_i{2} [id T] <TensorType(int64, scalar)> ''   
 | |   | |   |ctx_input [id U] <TensorType(float32, 3D)>
 | |   | |TensorConstant{354} [id V] <TensorType(int64, scalar)>
 | |   |Shape_i{1} [id W] <TensorType(int64, scalar)> ''   
 | |   | |lstm_1_W_i [id X] <TensorType(float32, matrix)>
 | |   |TensorConstant{30000} [id F] <TensorType(int64, scalar)>
 | |TensorConstant{300} [id Y] <TensorType(int64, scalar)>
 | |Shape_i{1} [id Z] <TensorType(int64, scalar)> ''   
 | | |lstm_2_W_i [id BA] <TensorType(float32, matrix)>
 | |TensorConstant{-120} [id BB] <TensorType(int64, scalar)>
 |TensorConstant{200} [id BC] <TensorType(int64, scalar)>

Storage map footprint:
 - Alloc.0, Shape: (100, 128, 1200), ElemSize: 4 Byte(s), TotalSize: 61440000 Byte(s)
 - Subtensor{int64:int64:int64}.0, Shape: (100, 128, 1200), ElemSize: 4 Byte(s), TotalSize: 61440000 Byte(s)
 - Reshape{2}.0, Shape: (12800, 354), ElemSize: 4 Byte(s), TotalSize: 18124800 Byte(s)
 - ctx_input, Input, Shape: (128, 100, 304), ElemSize: 4 Byte(s), TotalSize: 15564800 Byte(s)
 - Reshape{2}.0, Shape: (12800, 300), ElemSize: 4 Byte(s), TotalSize: 15360000 Byte(s)
 - Subtensor{int64:int64:int64}.0, Shape: (100, 128, 300), ElemSize: 4 Byte(s), TotalSize: 15360000 Byte(s)
 - Elemwise{tanh}.0, Shape: (100, 128, 300), ElemSize: 4 Byte(s), TotalSize: 15360000 Byte(s)
 - Elemwise{Composite{(i0 - sqr(i1))}}.0, Shape: (100, 128, 300), ElemSize: 4 Byte(s), TotalSize: 15360000 Byte(s)
 - Subtensor{int64:int64:int64}.0, Shape: (100, 128, 300), ElemSize: 4 Byte(s), TotalSize: 15360000 Byte(s)
 - InplaceDimShuffle{1,0,2}.0, Shape: (100, 128, 200), ElemSize: 4 Byte(s), TotalSize: 10240000 Byte(s)
 - forall_inplace,cpu,scan_fn}.1, Shape: (101, 128, 50), ElemSize: 4 Byte(s), TotalSize: 2585600 Byte(s)
 - Reshape{2}.0, Shape: (12800, 50), ElemSize: 4 Byte(s), TotalSize: 2560000 Byte(s)
 - Subtensor{int64:int64:int64}.0, Shape: (100, 128, 50), ElemSize: 4 Byte(s), TotalSize: 2560000 Byte(s)
 - Elemwise{tanh}.0, Shape: (100, 128, 50), ElemSize: 4 Byte(s), TotalSize: 2560000 Byte(s)
 - lstm_1_W_i, Shared Input, Shape: (354, 300), ElemSize: 4 Byte(s), TotalSize: 424800 Byte(s)
 - lstm_1_W_f, Shared Input, Shape: (354, 300), ElemSize: 4 Byte(s), TotalSize: 424800 Byte(s)
 - lstm_1_W_c, Shared Input, Shape: (354, 300), ElemSize: 4 Byte(s), TotalSize: 424800 Byte(s)
 - lstm_1_W_o, Shared Input, Shape: (354, 300), ElemSize: 4 Byte(s), TotalSize: 424800 Byte(s)
 - <TensorType(float32, matrix)>, Shared Input, Shape: (354, 300), ElemSize: 4 Byte(s), TotalSize: 424800 Byte(s)
 - <TensorType(float32, matrix)>, Shared Input, Shape: (354, 300), ElemSize: 4 Byte(s), TotalSize: 424800 Byte(s)
 - <TensorType(float32, matrix)>, Shared Input, Shape: (354, 300), ElemSize: 4 Byte(s), TotalSize: 424800 Byte(s)
 - <TensorType(float32, matrix)>, Shared Input, Shape: (354, 300), ElemSize: 4 Byte(s), TotalSize: 424800 Byte(s)
 - <TensorType(float32, matrix)>, Shared Input, Shape: (300, 300), ElemSize: 4 Byte(s), TotalSize: 360000 Byte(s)
 - lstm_1_U_o, Shared Input, Shape: (300, 300), ElemSize: 4 Byte(s), TotalSize: 360000 Byte(s)
 - lstm_1_U_f, Shared Input, Shape: (300, 300), ElemSize: 4 Byte(s), TotalSize: 360000 Byte(s)
 - lstm_1_U_i, Shared Input, Shape: (300, 300), ElemSize: 4 Byte(s), TotalSize: 360000 Byte(s)
 - <TensorType(float32, matrix)>, Shared Input, Shape: (300, 300), ElemSize: 4 Byte(s), TotalSize: 360000 Byte(s)
 - lstm_1_U_c, Shared Input, Shape: (300, 300), ElemSize: 4 Byte(s), TotalSize: 360000 Byte(s)
 - <TensorType(float32, matrix)>, Shared Input, Shape: (300, 300), ElemSize: 4 Byte(s), TotalSize: 360000 Byte(s)
 - <TensorType(float32, matrix)>, Shared Input, Shape: (300, 300), ElemSize: 4 Byte(s), TotalSize: 360000 Byte(s)
 - <TensorType(float32, matrix)>, Shared Input, Shape: (300, 50), ElemSize: 4 Byte(s), TotalSize: 60000 Byte(s)
 - lstm_2_W_f, Shared Input, Shape: (300, 50), ElemSize: 4 Byte(s), TotalSize: 60000 Byte(s)
 - lstm_2_W_c, Shared Input, Shape: (300, 50), ElemSize: 4 Byte(s), TotalSize: 60000 Byte(s)
 - <TensorType(float32, matrix)>, Shared Input, Shape: (300, 50), ElemSize: 4 Byte(s), TotalSize: 60000 Byte(s)
 - <TensorType(float32, matrix)>, Shared Input, Shape: (300, 50), ElemSize: 4 Byte(s), TotalSize: 60000 Byte(s)
 - <TensorType(float32, matrix)>, Shared Input, Shape: (300, 50), ElemSize: 4 Byte(s), TotalSize: 60000 Byte(s)
 - lstm_2_W_i, Shared Input, Shape: (300, 50), ElemSize: 4 Byte(s), TotalSize: 60000 Byte(s)
 - lstm_2_W_o, Shared Input, Shape: (300, 50), ElemSize: 4 Byte(s), TotalSize: 60000 Byte(s)
 - dep_input, Input, Shape: (128, 100), ElemSize: 4 Byte(s), TotalSize: 51200 Byte(s)
 - timedistributed_1_target, Input, Shape: (128, 100, 1), ElemSize: 4 Byte(s), TotalSize: 51200 Byte(s)
 - Elemwise{Cast{int32}}.0, Shape: (12800,), ElemSize: 4 Byte(s), TotalSize: 51200 Byte(s)
 - Elemwise{sub,no_inplace}.0, Shape: (128, 100, 1), ElemSize: 4 Byte(s), TotalSize: 51200 Byte(s)
 - embedding_1_W, Shared Input, Shape: (101, 50), ElemSize: 4 Byte(s), TotalSize: 20200 Byte(s)
 - <TensorType(float32, matrix)>, Shared Input, Shape: (101, 50), ElemSize: 4 Byte(s), TotalSize: 20200 Byte(s)
 - Alloc.0, Shape: (101, 50), ElemSize: 4 Byte(s), TotalSize: 20200 Byte(s)
 - <TensorType(float32, matrix)>, Shared Input, Shape: (50, 50), ElemSize: 4 Byte(s), TotalSize: 10000 Byte(s)
 - lstm_2_U_o, Shared Input, Shape: (50, 50), ElemSize: 4 Byte(s), TotalSize: 10000 Byte(s)
 - lstm_2_U_f, Shared Input, Shape: (50, 50), ElemSize: 4 Byte(s), TotalSize: 10000 Byte(s)
 - lstm_2_U_i, Shared Input, Shape: (50, 50), ElemSize: 4 Byte(s), TotalSize: 10000 Byte(s)
 - lstm_2_U_c, Shared Input, Shape: (50, 50), ElemSize: 4 Byte(s), TotalSize: 10000 Byte(s)
 - <TensorType(float32, matrix)>, Shared Input, Shape: (50, 50), ElemSize: 4 Byte(s), TotalSize: 10000 Byte(s)
 - <TensorType(float32, matrix)>, Shared Input, Shape: (50, 50), ElemSize: 4 Byte(s), TotalSize: 10000 Byte(s)
 - <TensorType(float32, matrix)>, Shared Input, Shape: (50, 50), ElemSize: 4 Byte(s), TotalSize: 10000 Byte(s)
 - lstm_1_b_i, Shared Input, Shape: (300,), ElemSize: 4 Byte(s), TotalSize: 1200 Byte(s)
 - lstm_1_b_f, Shared Input, Shape: (300,), ElemSize: 4 Byte(s), TotalSize: 1200 Byte(s)
 - lstm_1_b_c, Shared Input, Shape: (300,), ElemSize: 4 Byte(s), TotalSize: 1200 Byte(s)
 - <TensorType(float32, vector)>, Shared Input, Shape: (300,), ElemSize: 4 Byte(s), TotalSize: 1200 Byte(s)
 - <TensorType(float32, vector)>, Shared Input, Shape: (300,), ElemSize: 4 Byte(s), TotalSize: 1200 Byte(s)
 - <TensorType(float32, vector)>, Shared Input, Shape: (300,), ElemSize: 4 Byte(s), TotalSize: 1200 Byte(s)
 - lstm_1_b_o, Shared Input, Shape: (300,), ElemSize: 4 Byte(s), TotalSize: 1200 Byte(s)
 - <TensorType(float32, vector)>, Shared Input, Shape: (300,), ElemSize: 4 Byte(s), TotalSize: 1200 Byte(s)
 - timedistributed_1_sample_weights, Input, Shape: (128,), ElemSize: 4 Byte(s), TotalSize: 512 Byte(s)
 - <TensorType(float32, vector)>, Shared Input, Shape: (50,), ElemSize: 4 Byte(s), TotalSize: 200 Byte(s)
 - lstm_2_b_i, Shared Input, Shape: (50,), ElemSize: 4 Byte(s), TotalSize: 200 Byte(s)
 - lstm_2_b_f, Shared Input, Shape: (50,), ElemSize: 4 Byte(s), TotalSize: 200 Byte(s)
 - dense_1_W, Shared Input, Shape: (50, 1), ElemSize: 4 Byte(s), TotalSize: 200 Byte(s)
 - <TensorType(float32, matrix)>, Shared Input, Shape: (50, 1), ElemSize: 4 Byte(s), TotalSize: 200 Byte(s)
 - <TensorType(float32, vector)>, Shared Input, Shape: (50,), ElemSize: 4 Byte(s), TotalSize: 200 Byte(s)
 - <TensorType(float32, vector)>, Shared Input, Shape: (50,), ElemSize: 4 Byte(s), TotalSize: 200 Byte(s)
 - <TensorType(float32, vector)>, Shared Input, Shape: (50,), ElemSize: 4 Byte(s), TotalSize: 200 Byte(s)
 - lstm_2_b_c, Shared Input, Shape: (50,), ElemSize: 4 Byte(s), TotalSize: 200 Byte(s)
 - lstm_2_b_o, Shared Input, Shape: (50,), ElemSize: 4 Byte(s), TotalSize: 200 Byte(s)
 - TensorConstant{(4,) of 300}, Shape: (4,), ElemSize: 8 Byte(s), TotalSize: 32 Byte(s)
 - TensorConstant{(4,) of 50}, Shape: (4,), ElemSize: 8 Byte(s), TotalSize: 32 Byte(s)
 - MakeVector{dtype='int64'}.0, Shape: (3,), ElemSize: 8 Byte(s), TotalSize: 24 Byte(s)
 - MakeVector{dtype='int64'}.0, Shape: (3,), ElemSize: 8 Byte(s), TotalSize: 24 Byte(s)
 - MakeVector{dtype='int64'}.0, Shape: (3,), ElemSize: 8 Byte(s), TotalSize: 24 Byte(s)
 - MakeVector{dtype='int64'}.0, Shape: (2,), ElemSize: 8 Byte(s), TotalSize: 16 Byte(s)
 - MakeVector{dtype='int64'}.0, Shape: (2,), ElemSize: 8 Byte(s), TotalSize: 16 Byte(s)
 - MakeVector{dtype='int64'}.0, Shape: (2,), ElemSize: 8 Byte(s), TotalSize: 16 Byte(s)
 - MakeVector{dtype='int64'}.0, Shape: (2,), ElemSize: 8 Byte(s), TotalSize: 16 Byte(s)
 - MakeVector{dtype='int64'}.0, Shape: (2,), ElemSize: 8 Byte(s), TotalSize: 16 Byte(s)
 - MakeVector{dtype='int64'}.0, Shape: (2,), ElemSize: 8 Byte(s), TotalSize: 16 Byte(s)
 - MakeVector{dtype='int64'}.0, Shape: (2,), ElemSize: 8 Byte(s), TotalSize: 16 Byte(s)
 - MakeVector{dtype='int64'}.0, Shape: (2,), ElemSize: 8 Byte(s), TotalSize: 16 Byte(s)
 - TensorConstant{[50 -1]}, Shape: (2,), ElemSize: 8 Byte(s), TotalSize: 16 Byte(s)
 - MakeVector{dtype='int64'}.0, Shape: (2,), ElemSize: 8 Byte(s), TotalSize: 16 Byte(s)
 - TensorConstant{[300  -1]}, Shape: (2,), ElemSize: 8 Byte(s), TotalSize: 16 Byte(s)
 - MakeVector{dtype='int64'}.0, Shape: (2,), ElemSize: 8 Byte(s), TotalSize: 16 Byte(s)
 - MakeVector{dtype='int64'}.0, Shape: (2,), ElemSize: 8 Byte(s), TotalSize: 16 Byte(s)
 - TensorConstant{[ -1 100   1]}, Shape: (3,), ElemSize: 4 Byte(s), TotalSize: 12 Byte(s)
 - TensorConstant{[-1 50]}, Shape: (2,), ElemSize: 4 Byte(s), TotalSize: 8 Byte(s)
 - Constant{-1}, Shape: (), ElemSize: 8 Byte(s), TotalSize: 8.0 Byte(s)
 - TensorConstant{1200}, Shape: (), ElemSize: 8 Byte(s), TotalSize: 8.0 Byte(s)
 - Constant{-101}, Shape: (), ElemSize: 8 Byte(s), TotalSize: 8.0 Byte(s)
 - Elemwise{Composite{(((i0 // i1) * i2) // i3)}}.0, Shape: (), ElemSize: 8 Byte(s), TotalSize: 8.0 Byte(s)
 - TensorConstant{[ -1 354]}, Shape: (2,), ElemSize: 4 Byte(s), TotalSize: 8 Byte(s)
 - TensorConstant{200}, Shape: (), ElemSize: 8 Byte(s), TotalSize: 8.0 Byte(s)
 - TensorConstant{1}, Shape: (), ElemSize: 8 Byte(s), TotalSize: 8.0 Byte(s)
 - TensorConstant{-300}, Shape: (), ElemSize: 8 Byte(s), TotalSize: 8.0 Byte(s)
 - TensorConstant{[ -1 300]}, Shape: (2,), ElemSize: 4 Byte(s), TotalSize: 8 Byte(s)
 - TensorConstant{300}, Shape: (), ElemSize: 8 Byte(s), TotalSize: 8.0 Byte(s)
 - TensorConstant{5000}, Shape: (), ElemSize: 8 Byte(s), TotalSize: 8.0 Byte(s)
 - Elemwise{Composite{Switch(EQ(i0, i1), ((i2 * i0) // (i3 * i0)), i0)}}.0, Shape: (), ElemSize: 8 Byte(s), TotalSize: 8.0 Byte(s)
 - Constant{0}, Shape: (), ElemSize: 8 Byte(s), TotalSize: 8.0 Byte(s)
 - Elemwise{Composite{Switch(EQ(i0, i1), ((i2 * i0) // (i3 * i0)), i0)}}[(0, 0)].0, Shape: (), ElemSize: 8 Byte(s), TotalSize: 8.0 Byte(s)
 - Constant{2}, Shape: (), ElemSize: 8 Byte(s), TotalSize: 8.0 Byte(s)
 - Constant{1}, Shape: (), ElemSize: 8 Byte(s), TotalSize: 8.0 Byte(s)
 - TensorConstant{100}, Shape: (), ElemSize: 8 Byte(s), TotalSize: 8.0 Byte(s)
 - Constant{-102}, Shape: (), ElemSize: 8 Byte(s), TotalSize: 8.0 Byte(s)
 - Elemwise{mul,no_inplace}.0, Shape: (), ElemSize: 8 Byte(s), TotalSize: 8.0 Byte(s)
 - Elemwise{mul,no_inplace}.0, Shape: (), ElemSize: 8 Byte(s), TotalSize: 8.0 Byte(s)
 - TensorConstant{354}, Shape: (), ElemSize: 8 Byte(s), TotalSize: 8.0 Byte(s)
 - Constant{100}, Shape: (), ElemSize: 8 Byte(s), TotalSize: 8.0 Byte(s)
 - TensorConstant{30000}, Shape: (), ElemSize: 8 Byte(s), TotalSize: 8.0 Byte(s)
 - Constant{99}, Shape: (), ElemSize: 8 Byte(s), TotalSize: 8.0 Byte(s)
 - TensorConstant{50}, Shape: (), ElemSize: 8 Byte(s), TotalSize: 8.0 Byte(s)
 - TensorConstant{101}, Shape: (), ElemSize: 8 Byte(s), TotalSize: 8.0 Byte(s)
 - TensorConstant{-50}, Shape: (), ElemSize: 8 Byte(s), TotalSize: 8.0 Byte(s)
 - TensorConstant{-120}, Shape: (), ElemSize: 8 Byte(s), TotalSize: 8.0 Byte(s)
 - TensorConstant{[ -1 100 300]}, Shape: (3,), ElemSize: 2 Byte(s), TotalSize: 6 Byte(s)
 - TensorConstant{(1, 1, 1) of 0.02}, Shape: (1, 1, 1), ElemSize: 4 Byte(s), TotalSize: 4 Byte(s)
 - mean, Shape: (), ElemSize: 4 Byte(s), TotalSize: 4.0 Byte(s)
 - dense_1_b, Shared Input, Shape: (1,), ElemSize: 4 Byte(s), TotalSize: 4 Byte(s)
 - TensorConstant{0.009999999776482582}, Shape: (), ElemSize: 4 Byte(s), TotalSize: 4.0 Byte(s)
 - TensorConstant{(1, 1, 1) of 1.0}, Shape: (1, 1, 1), ElemSize: 4 Byte(s), TotalSize: 4 Byte(s)
 - <TensorType(float32, scalar)>, Shared Input, Shape: (), ElemSize: 4 Byte(s), TotalSize: 4.0 Byte(s)
 - <TensorType(float32, scalar)>, Shared Input, Shape: (), ElemSize: 4 Byte(s), TotalSize: 4.0 Byte(s)
 - <TensorType(float32, vector)>, Shared Input, Shape: (1,), ElemSize: 4 Byte(s), TotalSize: 4 Byte(s)
 - <TensorType(float32, scalar)>, Shared Input, Shape: (), ElemSize: 4 Byte(s), TotalSize: 4.0 Byte(s)
 - TensorConstant{1.0}, Shape: (), ElemSize: 4 Byte(s), TotalSize: 4.0 Byte(s)
 - <TensorType(float32, scalar)>, Shared Input, Shape: (), ElemSize: 4 Byte(s), TotalSize: 4.0 Byte(s)
 - mean, Shape: (), ElemSize: 4 Byte(s), TotalSize: 4.0 Byte(s)
 - Sum{acc_dtype=float64}.0, Shape: (), ElemSize: 4 Byte(s), TotalSize: 4.0 Byte(s)
 - TensorConstant{0.0}, Shape: (), ElemSize: 4 Byte(s), TotalSize: 4.0 Byte(s)
 - Subtensor{int64}.0, Shape: (), ElemSize: 4 Byte(s), TotalSize: 4.0 Byte(s)
 - TensorConstant{(1, 1) of 0.0}, Shape: (1, 1), ElemSize: 4 Byte(s), TotalSize: 4 Byte(s)
 - TensorConstant{(1, 1, 1) of 0.0}, Shape: (1, 1, 1), ElemSize: 4 Byte(s), TotalSize: 4 Byte(s)
 - InplaceDimShuffle{x,x}.0, Shape: (1, 1), ElemSize: 4 Byte(s), TotalSize: 4 Byte(s)
 - TensorConstant{(1,) of -1}, Shape: (1,), ElemSize: 4 Byte(s), TotalSize: 4 Byte(s)
 - Elemwise{Cast{float32}}.0, Shape: (), ElemSize: 4 Byte(s), TotalSize: 4.0 Byte(s)
 - Elemwise{Cast{float32}}.0, Shape: (), ElemSize: 4 Byte(s), TotalSize: 4.0 Byte(s)
 - InplaceDimShuffle{x,x}.0, Shape: (1, 1), ElemSize: 4 Byte(s), TotalSize: 4 Byte(s)
 - TensorConstant{[ -1 100  50]}, Shape: (3,), ElemSize: 1 Byte(s), TotalSize: 3 Byte(s)
 - TensorConstant{300}, Shape: (), ElemSize: 2 Byte(s), TotalSize: 2.0 Byte(s)
 - TensorConstant{2}, Shape: (), ElemSize: 1 Byte(s), TotalSize: 1.0 Byte(s)
 - TensorConstant{(1,) of 0}, Shape: (1,), ElemSize: 1 Byte(s), TotalSize: 1 Byte(s)
 - TensorConstant{1}, Shape: (), ElemSize: 1 Byte(s), TotalSize: 1.0 Byte(s)
 - TensorConstant{-1}, Shape: (), ElemSize: 1 Byte(s), TotalSize: 1.0 Byte(s)
 - TensorConstant{50}, Shape: (), ElemSize: 1 Byte(s), TotalSize: 1.0 Byte(s)
 TotalSize: 260991796.0 Byte(s) 0.243 GB
 TotalSize inputs: 22558484.0 Byte(s) 0.021 GB

```

**Comment by nouiz** [comment]
the problem is that Theano tries to instantiate a 3d tensor of shape: 100,
-5334, 200.

negative shapes aren't good.

You probably don't run the same version of the code in both places. Verify
that.

**Comment by munichong** [comment]
@nouiz Thanks for your comment. I checked the code. I still do not know where -5334 comes from. The code is exactly the same as the one that I am using on the Linux laptop.

The problem is solved after I upgrade Theano from 0.8.2 to the Github dev version.
